### DISCUSSION TITLE
Is it possible to add third-party statistical information to the URL?

### DISCUSSION THREAD

**Discussion by wyfang** [YOURLS/YOURLS#2901] [Q&A] [answered]
Usually in my personal html, I will add such a statistic:  `<script src="/analytics.js"></script>`

Among them, I have two statistics, one Google Analytics and the other Baidu statistics：
`
//baidu tong ji
var _hmt = _hmt || [];
(function() {
var hm = document.createElement("script");
hm.src = "https://hm.baidu.com/hm.js?c1c28557921a35daafd53672a70cb877";
var s = document.getElementsByTagName("script")[0]; 
s.parentNode.insertBefore(hm, s);
})();

//google
var s = document.createElement('script');
s.async = true;
s.src = 'https://www.googletagmanager.com/gtag/js?id=UA-117490728-1';
document.head.appendChild(s);
window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-117490728-1');

//baidu
var _mtac = {};
(function() {
var mta = document.createElement("script");
mta.src = "//pingjs.qq.com/h5/stats.js?v2.0.4";
mta.setAttribute("name", "MTAH5");
mta.setAttribute("sid", "500679308");
var s = document.getElementsByTagName("script")[0];
s.parentNode.insertBefore(mta, s);
})();
`

Therefore, I hope that the link shared with others after shortening the url can also be added to my statistics. What should I do?

**Comment by ozh** [comment, marked as answer]
These stats tools need an HTML page to be displayed, so if you want your short URL to use them, you need a plugin that will load a page with these stat scripts before redirecting to the long URL.

See https://github.com/YOURLS/YOURLS/wiki/Possible-With-A-Plugin
